Xamarin.PropertyEditing.Mac/Controls/BasePathEditorControl.cs

```diff
@@ -23,11 +23,35 @@ internal abstract class BasePathEditorControl<T> : PropertyEditorControl<Propert
 		private readonly NSObject[] objects;
 		public override NSObject[] AccessibilityChildren { get => this.objects; }
 
+		private readonly TextNextResponderDelegate textNextResponderDelegate;
+
+		class BasePathEditorDelegate : TextNextResponderDelegate
+		{
+			WeakReference<BasePathEditorControl<T>> weakView;
+
+			public BasePathEditorDelegate (BasePathEditorControl<T> basePathEditorControl)
+			{
+				weakView = new WeakReference<BasePathEditorControl<T>>(basePathEditorControl);
+			}
+
+			public override void Changed (NSNotification notification)
+			{
+				if (this.weakView.TryGetTarget(out BasePathEditorControl<T> t)) {
+					t.StoreCurrentValue ();
+				}
+			}
+		}
+
 		protected BasePathEditorControl (IHostResourceProvider hostResource)
 			: base (hostResource)
 		{
 			this.currentTextField = new TextFieldSmallButtonContainer ();
-			this.currentTextField.Changed += CurrentTextField_Changed;
+
+			this.textNextResponderDelegate = new BasePathEditorDelegate (this)
```

Comment:
Pass the ProxyResponder as a parameter. Or just construct it in the delegate, you have `this` anyway.

Reply:
I want to do another review and improving it, I will include the changes in your comments on it.